ksql-serde/src/main/java/io/confluent/ksql/serde/json/KsqlJsonSerializer.java

```diff
@@ -47,27 +54,156 @@ public void configure(final Map<String, ?> props, final boolean isKey) {
     if (data == null) {
       return null;
     }
-
     try {
-      return objectMapper.writeValueAsBytes(dataToMap(data));
+      Struct struct = new Struct(schema);
+      for (int i = 0; i < data.getColumns().size(); i++) {
+        if (schema.fields().get(i).schema().type() == Schema.Type.STRUCT) {
+          struct.put(schema.fields().get(i), updateStructSchema(
```

Comment:
The purpose of the call to updateStructSchema (to ensure we pass validation in struct.put) is very subtle. we should add a comment explaining that updateStructSchema isn't actually doing anything to the data and just sets the schema up so that put doesn't fail.